zkevm-circuits/src/evm_circuit/execution/precompiles/sha256.rs

```diff
@@ -78,7 +78,11 @@ impl<F: Field> ExecutionGadget<F> for SHA256Gadget<F> {
 
         // sha256 verify lookup
         cb.condition(is_success.expr(), |cb| {
-            cb.sha256_table_lookup(input_bytes_rlc.expr(), output_bytes_rlc.expr());
+            cb.sha256_table_lookup(
+                input_bytes_rlc.expr(),
+                call_data_length.expr(),
+                output_bytes_rlc.expr(),
+            );
```

Comment:
We should add a test case with `call_data_length == 0`, i.e. `input_len == 0`. In this case, the precompiled contract call will be successful, so a lookup to the keccak table will be done. However, we must ensure this "empty" input case is handled from the SHA256 circuit (that it was populated as witness, and not ignored)

Reply:
done with a dedicated unittest